src/troubleshooting.rst

```diff
@@ -0,0 +1,204 @@
+   CYLC_JOB_INIT_TIME=2000-01-01T00:10:00
+   CYLC_JOB_EXIT=SUCCEEDED
+   CYLC_JOB_EXIT_TIME=2000-01-01T01:30:00
+
+If the ``job.status`` file is showing something different to what the GUI or
+Tui is showing, then...
+
+.. rubric:: If your platform uses push communication:
+
+If messages aren't getting back to the scheduler, there should be some
+evidence of this in the ``job.err`` file, likely either an error or a
+traceback.
+
+Likely causes:
+
+* There is a network issue.
+* TCP ports are not open (zmq communications).
+* Non-interactive SSH has not been correctly configured (ssh communications).
+
+.. rubric:: If your platform uses pull communication:
+
+Firstly, check the polling interval, it's possible that the scheduler has been
+configured to poll infrequently and you need to wait for the next poll, or use
+the ``cylc poll`` command (also available in the GUI).
+
+Use the ``cylc config`` command to inspect the platform's configuration to
+determine the configured polling schedule.
+
+Then check the ``job-activity.log`` file, there may have been a problem polling
+the remote platform, e.g. a network or configuration error.
+
+Likely causes:
+
+* The platform is down (e.g. all login nodes are offline).
+* There is a network issue.
+* Non-interactive SSH has not been correctly configured.
+
+
+My Job Submit-Failed
+^^^^^^^^^^^^^^^^^^^^
+
+A submit-failed job means one of three things:
+
+1. There is a Bash syntax error in the task configuration.
+
+   E.G. the following ``script`` has a syntax error, it is missing a
+   ``"`` character:
+   
+   .. code-block:: cylc
+   
+      [runtime]
+          [[foo]]
+              script = echo "Hello $WORLD
+   
+   This will result in a submission-failure which should appear in the
+   ``job-activity.log`` file (and also the scheduler log) something like this:
+   
+   .. code-block::
+   
+      /path/to/job.tmp: line 46: unexpected EOF while looking for matching `"'
+      /path/to/job.tmp: line 50: syntax error: unexpected end of file
+
+2. There was an error submitting the job to the specified platform (including
+   network issues).
+
+   See the ``job-activity.log`` and the scheduler log. The error should be in
+   one or both of those files.
+
+3. The platform is not correctly configured.
+
+
+My Job Failed
+^^^^^^^^^^^^^
+
+This means something went wrong executing the job.
+
+To find out more, see the ``job.err`` file.
+
+If you're struggling to track down the error, you might want to restart the
+workflow in debug mode and run the task again:
+
+.. code-block:: console
+
+   # shut the workflow down (leave any active jobs running)
+   $ cylc stop --now --now <workflow>
+   # restart the workflow in debug mode
+   $ cylc play <workflow> --debug
+   # re-run all failed task(s)
+   $ cylc trigger '<workflow>//*:failed'
+
+When a workflow is running in debug mode, all jobs will create a ``job.xtrace``
+file which can help you to locate the error within the job script.
+
```

Comment:
Worth shoehorning a note about `set -eu` in here? Something like:

```suggestion
.. note::

   "My job didn't fail, but should have": 
   
   Bash scripts do not by default fail on error. Use `set -e` to alter this default. 
   
   See :ref:`workflow_design_guide.basic_principles.basic_functionality`

```

Reply:
(Should mention that Cylc job scripts do use `set -eu`, so this advice applies to external scripts called by the job)